### PR TITLE
Daffodil 1856 - Updates to for Apache incubation

### PR DESCRIPTION
**Pull request by stevedlawrence**
This is split up into 4 commits to make things easier to review:

1. Remove old files
2. Move the directory layout. All changes are a 100% rename
3. Update package and xml namespaces to be apache based, adding backwards compatability for namespaces
4. Changes to add "Apache" and "incubating" where necessary, including updating the version to include "-incubating" and to be manually updated.

### CODE REVIEW

File: daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala [hunk 1]
```diff
@@ -1302,13 +1302,22 @@ case class FunctionCallExpression(functionQNameString: String, expressions: List
...
+        DAFTraceExpr(functionQNameString, functionQName, args)
+
+      case (RefQName(_, "error", DAF_NCSA), args) => {
+        SDW("Expression daf:error is deprecated. Use fn:error instead")
+        DAFErrorExpr(functionQNameString, functionQName, args)
+      }
+
+      case (RefQName(_, "error", DAF_APACHE), args) => {
         SDW("Expression daf:error is deprecated. Use fn:error instead")
         DAFErrorExpr(functionQNameString, functionQName, args)
```
Comment (stevedlawrence) [inline daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala]: Changes to support xml namespace backwards compatibility.

File: daffodil-core/src/main/scala/org/apache/daffodil/dsom/IIBase.scala [hunk 2]
```diff
@@ -179,7 +179,17 @@ abstract class IIBase(xml: Node, xsdArg: XMLSchemaDocument, val seenBefore: IIMa
...
+    val prop = getAttributeOption("schemaLocation")
+    prop.map { text =>
+      if (text.contains("edu/illinois/ncsa/daffodil")) {
+        SDW("schemaLocation property uses deprecated edu/illinois/ncsa/daffodil path instead of org/apache/daffodil. Converting to new path.")
+        text.replace("edu/illinois/ncsa/daffodil", "org/apache/daffodil")
+      } else {
+        text
+      }
+    }
+  }
```
Comment (stevedlawrence) [inline daffodil-core/src/main/scala/org/apache/daffodil/dsom/IIBase.scala]: Supports backwards compatibility for schemas that still reference the old paths. For example, it is common to have an import with schemaLocation="edu/illinois/ncsa/daffodil/built-in-formats.xsd"

**Comment by johnament** [comment]
Please note - the requirement is to have the word "incubating" in the source release.  There is no requirement that the version # has to include "incubating" but most people find this the easiest way to handle it.  It has the downside that your convenience binaries now unnecessarily include "-incubating" in them.

File: build.sbt [hunk 3]
```diff
@@ -21,7 +23,7 @@ testOptions in ThisBuild += Tests.Argument(TestFrameworks.JUnit, "-v")
 
 transitiveClassifiers := Seq("sources", "javadoc")
 
-resolvers in ThisBuild += "NCSA Sonatype Releases" at "https://opensource.ncsa.illinois.edu/nexus/content/repositories/releases"
+resolvers in ThisBuild += "NCSA Sonatype Releases" at "https://daffodil.apache.org/nexus/content/repositories/releases"
```
Comment (johnament) [inline build.sbt]: This url doesn't work.

File: daffodil-cli/DISCLAIMER [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Apache Daffodil is an effort undergoing incubation at The Apache Software
```
Comment (johnament) [inline daffodil-cli/DISCLAIMER]: The disclaimer should only have to appear once, in the source root.  If this is the root of the binary artifact that's fine, but its ideal if it can be a single entry.

File: daffodil-lib/src/main/scala/org/apache/daffodil/xml/XMLUtils.scala [hunk 5]
```diff
@@ -339,20 +339,24 @@ object XMLUtils {
...
+  val EXT_PREFIX_NCSA = "daf"
+  val EXT_NS_NCSA = NS(DAFFODIL_EXTENSION_NAMESPACE_NCSA.uri)
+
+  private val DAFFODIL_EXTENSIONS_NAMESPACE_ROOT_APACHE = "urn:ogf:dfdl:2013:imp:daffodil.apache.org:2017"
+  private val DAFFODIL_EXTENSION_NAMESPACE_APACHE = NS(DAFFODIL_EXTENSIONS_NAMESPACE_ROOT_APACHE + ":ext")
+  val EXT_PREFIX_APACHE = "daf"
+  val EXT_NS_APACHE = NS(DAFFODIL_EXTENSION_NAMESPACE_APACHE.uri)
+
+  private val DAFFODIL_INTERNAL_NAMESPACE = NS(DAFFODIL_EXTENSIONS_NAMESPACE_ROOT_APACHE + ":int")
+  val INT_PREFIX= "dafint"
```
Comment (stevedlawrence) [inline daffodil-lib/src/main/scala/org/apache/daffodil/xml/XMLUtils.scala]: New XML Namespace defined.  Support for the NCSA namespace is still provided for backwards compatibility. The internal namespace does not maintain comapt, since it's internal only.

File: daffodil-cli/src/main/scala/org/apache/daffodil/debugger/CLIDebuggerRunner.scala [hunk 6]
```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2012-2015 Tresys Technology, LLC. All rights reserved.
```
Comment (johnament) [inline daffodil-cli/src/main/scala/org/apache/daffodil/debugger/CLIDebuggerRunner.scala]: We'll typically only change package name once the SGA is on file and we can remove copyright.

File: build.sbt (hunk 3 above)
Comment (stevedlawrence) [reply build.sbt]: Actually, this line doesn't even belong in here anymore. Daffodil does not require dependencies from anything on NCSA. I'll remove this line.

File: daffodil-cli/DISCLAIMER (hunk 4 above)
Comment (stevedlawrence) [reply daffodil-cli/DISCLAIMER]: This was done because the daffodil-cli directory is a subproject, and there are some cases where SBT will prevent one from pulling in files outside of a subdirectory. This might not be one of those cases, I'll see if there is a way to only have one file.

File: daffodil-cli/src/main/scala/org/apache/daffodil/debugger/CLIDebuggerRunner.scala (hunk 6 above)
Comment (stevedlawrence) [reply daffodil-cli/src/main/scala/org/apache/daffodil/debugger/CLIDebuggerRunner.scala]: Ok. So maybe we should hold off on the two commits that change the directory layout and package names/XML namespaces for now.

**Comment by stevedlawrence** [comment]
Regarding the "-incubating" comment in the version, I was just following what other projects seemed to do. If it's only required in the source release, I'd probably prefer to not include it in the version. It added a little extra complexity and makes the version numbers a bit messy.

**Comment by mbeckerle** [comment]
I'd really like to schedule this change specifically. I.e., coordinate so all outstanding work is committed, then this change occurs. Because I have outstanding changes that include removal and renaming of files, movement of code from class to class and file to file. 

Past experience has told me that major rename/relabel events need to happen without putting a huge burden on those with major sets of outstanding changes. 

In other words. I'd like 60 days notice before this happens.

**Comment by stevedlawrence** [comment]
Seems reasonable. I'll close this pull request and create a new one that just adds the Apache incubator stuff and a few other minor things, plus addresses the comments in this thread.
